src/pages/products/ProductCards.jsx

```diff
@@ -1,27 +1,17 @@
 import styles from "./Products.module.css";
-import Hat from "../../assets/card-img1.png";
-import React, { useState, useEffect } from "react";
-import axios from "axios";
+import React from "react";
+import { useProducts } from "../../utils/product-context";
 
 export default function ProductCards() {
-  const [products, setProducts] = useState([]);
+  const { finalProducts } = useProducts();
 
-  async function getProducts() {
-    try {
-      const res = await axios.get("/api/products");
-      setProducts(res.data.products);
-    } catch (error) {
-      console.log(error);
-    }
-  }
-
-  useEffect(() => {
-    getProducts();
-  }, []);
   return (
     <section
       className={`min-h-100 grid ${styles.grid__cols__auto} ${styles.grid__col__1} ${styles.grid__cols__2} gap-2 my-3 px-md`}>
-      {products.map((product) => (
+      {finalProducts.length === 0 && (
+        <div className="h1">No Products Available :(</div>
+      )}
+      {finalProducts.map((product) => (
         <div
```

Comment:
Used es5 functions as it sacrifices modernity for better readibility, it might differ from person to person but I feel it makes code reading and review much easier

As for utils and components mentioned, yes that can definitely be improved in upcoming PRs